### PR TITLE
Add scaffolding for the framegraph feature

### PR DESCRIPTION
**Pull request by hevrard**
This adds the scaffolding code to start working on the framegraph
feature: a gapit framegraph verb and a Framegraph path to be resolved
by a Get RPC.

The code here always returns an empty framegraph: this is on purpose,
the actual creation of a meaningful framegraph will be added in
subsequent changes, once this scaffolding is in place.

Bug: b/171694399

### CODE REVIEW

**Review by johnplate — CHANGES_REQUESTED** [review]
The kokoro test failed:
API does not implement api.API (missing GetFramegraph method)

**Comment by hevrard** [comment]
OK, I fixed the tests.

File: gapis/service/path/path.proto
Comment (yalcinmelihyasin) [inline gapis/service/path/path.proto:47]: Style: I am not sure about this id. We update client and server at the same time. Therefore changing the number wouldn't be an issue(except the unnecessary change of 20 lines).

But outside of that we have a unusual usage of protobuff field numbers here. We put them in alphabetical order and chronological order(the numbers). As we easily think that we cannot guarantee that every new field will be continued to named alphabetically, just as in here, breaking this is inevitable. But, in my opinion, the more important one of those is the numbers instead of alphabetical order. So rather than breaking the number order, I would break the alphabetical order. Because every time we have to add a new field here, we would need to check all the numbers and ensure we don't have the same number. Because they have to be unique. But keeping the alphabetical order does not give anything to us.

File: gapis/api/vulkan/framegraph.go
Comment (yalcinmelihyasin) [inline gapis/api/vulkan/framegraph.go:28]: nit: I may choose returning nil instead of an empty struct.

File: gapis/service/path/path.proto
Comment (hevrard) [reply gapis/service/path/path.proto]: Yes, thanks for spotting this! I was trying to at the same time not rewrite IDs and keep the alphabetical order, but as you say that approach would lead to something where it's difficult to know the current max ID.

I've moved this to the bottom of the list.

File: gapis/api/vulkan/framegraph.go
Comment (hevrard) [reply gapis/api/vulkan/framegraph.go]: Done

**Review by johnplate — APPROVED** [review]
LGTM, but wait for at least one more approval.

File: cmd/gapit/framegraph.go
Comment (yalcinmelihyasin) [inline cmd/gapit/framegraph.go:92]: nit: Can this be reduced into a string format? e.g `fmt.sprintf("...`

Comment (yalcinmelihyasin) [reply cmd/gapit/framegraph.go:92]: I think that it would be much readable.

File: cmd/gapit/framegraph.go
Comment (yalcinmelihyasin) [inline cmd/gapit/framegraph.go:85]: A general feedback about this: 

Golang strings are immutable and every time there is a an add operation, it may trigger a new allocation. So, this does not seem to be a hot path call, it's probably a call that runs once per query so not a big issue but if you plan to do more work on this in the future, it may be good to look at if this can be converted to something easier to read, more efficient to run with either string formatting or other alternatives.

**Review by yalcinmelihyasin — APPROVED** [review]

File: cmd/gapit/framegraph.go
Comment (hevrard) [reply cmd/gapit/framegraph.go:85]: Thanks for the feedback!

I am aware of the perf impact of the string manipulations here, I used it for prototyping and knowing this won't be noticeable given that we deal with not-very-big files here. (To give an idea, generating the dependency graph is in the order of several seconds, so in comparison producing the framegraph (which is much smaller than the dependency graph) dot representation is negligible.)

I'll keep your remarks in mind and improve once we have a working framegraph!

File: cmd/gapit/framegraph.go
Comment (hevrard) [reply cmd/gapit/framegraph.go:92]: WIll adress in future PRs once we have a working framegraph MVP.

File: gapis/api/api.go
Comment (pmuetschard) [inline gapis/api/api.go:63]: Since you are resolving the capture in order to get the API instance, passing in the capture path here is useless, please remove it.

File: gapis/resolve/framegraph.go
Comment (pmuetschard) [inline gapis/resolve/framegraph.go:28]: This is not necessary. Anything that is returned via the `Get` API is already stored in the database. Also, since you are calling `database.Build`, it's run in-place, so just put the code from the below `Resolve` here.

File: gapis/api/api.go
Comment (hevrard) [reply gapis/api/api.go:63]: The path is needed in the implementation of `GetFramegraph` for two things:
- create the dependency graph
- sync.MutateWithSubcommands

Do you prefer to pass the capture and re-create its path inside `GetFramegraph`?

File: gapis/resolve/framegraph.go
Comment (hevrard) [reply gapis/resolve/framegraph.go:28]: Fixed with: https://github.com/google/agi/pull/542

Note that I was mirroring `gapis/resolve/resources.go`, which has this double-function layout .

**Pull request merged**
